Solutions/Menes.Specs/Steps/OpenApiValidationSteps.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace Menes.Specs.Steps
+{
+    using System;
+    using Corvus.SpecFlow.Extensions;
+    using Menes.Exceptions;
+    using Menes.Validation;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.OpenApi;
+    using Microsoft.OpenApi.Models;
+    using Microsoft.OpenApi.Readers;
+    using Newtonsoft.Json.Linq;
+    using NUnit.Framework;
+    using TechTalk.SpecFlow;
+
+    [Binding]
+    public class OpenApiValidationSteps
+    {
+        private readonly ScenarioContext scenarioContext;
+
+        public OpenApiValidationSteps(ScenarioContext scenarioContext)
+        {
+            this.scenarioContext = scenarioContext;
+        }
+
+        [Given("the schema '(.*)'")]
+        public void GivenTheSchema(string schema)
+        {
+            var reader = new OpenApiStringReader();
+            OpenApiSchema openApiSchema = reader.ReadFragment<OpenApiSchema>(schema, OpenApiSpecVersion.OpenApi3_0, out OpenApiDiagnostic diagnostic);
+
+            if (diagnostic.Errors.Count > 0)
+            {
+                throw new ArgumentException("Invalid schema", nameof(schema));
+            }
+
+            this.scenarioContext.Set(openApiSchema);
+        }
+
+        [Given("the payload '(.*)'")]
+        public void GivenThePayload(string payload)
+        {
+            this.scenarioContext.Set(JToken.Parse(payload));
+        }
+
+        [When("I validate the payload against the schema")]
+        public void WhenIValidateThePayloadAgainstTheSchema()
+        {
+            OpenApiSchema schema = this.scenarioContext.Get<OpenApiSchema>();
+            JToken payload = this.scenarioContext.Get<JToken>();
+
+            OpenApiSchemaValidator validator = ContainerBindings.GetServiceProvider(this.scenarioContext).GetService<OpenApiSchemaValidator>();
+            try
+            {
+                validator.ValidateAndThrow(payload, schema);
+                this.scenarioContext.Set(true, "Result");
+            }
+            catch(OpenApiBadRequestException ex)
+            {
+                this.scenarioContext.Set(false, "Result");
```

Comment:
We should probably use a constant instead of the "Result" magic string, since it's being used in multiple places.

Reply:
Done